### DISCUSSION TITLE
How to get current walletClient chain id?

### DISCUSSION THREAD

**Discussion by alexbakers** [wevm/viem#378] [Question] [answered]
Hello, friends.

https://viem.sh/docs/actions/public/getChainId.html

```javascript
const chain = await publicClient.getChainId() 
```

How to get current walletClient chain id?

```javascript
const chain = await walletClient.getChainId() 
if (chain !== 1) {
    await walletClient.switchChain(mainnet);
}
```

**Comment by alexbakers** [comment, marked as answer]
Ohh, it's works, please add to documentation.

**Comment by tmm** [comment]
`getChainId` fetches the chain id from the network. Can also do the following to get the chain id for the specific client (if chain is configured):

```ts
import { createWalletClient, custom } from 'viem'
import { mainnet } from 'viem/chains'

const client = createWalletClient({
  chain: mainnet, // <--------- chain is configured
  transport: custom(window.ethereum!),
})
client.chain.id
//           ^? (property) id: 1
```

[TypeScript Playground](https://www.typescriptlang.org/play?#code/JYWwDg9gTgLgBAbzgYygUwIYzQdQwG3zRgGF9g0A7GAGhQFcBnGCEOAXzgDMpW4ByAG4UQ-AFChIsRHBAZglSsQ7debISID0yABbzKjcWOQQD8ZOSrwAvCnRZcBIqUvUAFAjFwUehQC5ZfSVaLzgYKAwDKRgA5CYWEDcAdwUAEwgkgDpiHTR0ehAAQgBKGjF2YuNXGEzdfUzgVLFNTW829u8APQB+MSA)

**Reply by alexbakers** [reply]
Why were two clients made, walletClient and publicClient?

Because they have different methods, it's very confusing. I would like to use only walletClient for example, but there is no method such as multicall.

**Reply by jxom** [reply]
They are separated because it is not guaranteed that a wallet’s EIP1193 provider has access to some public methods like eth_call.

For instance, some browser extension wallets (& WalletConnect v2) might not permit the eth_call method, so calling multicall will not work.

**Reply by chybisov** [reply]
@jxom is there any neat way to check those methods are permitted?

**Reply by jxom** [reply]
See if the wallet throws an error when you call the method 😀
